modules/dnn/src/model.cpp

```diff
@@ -488,307 +489,669 @@ void DetectionModel::detect(InputArray frame, CV_OUT std::vector<int>& classIds,
+{
+    return TextDetectionModel_EAST_Impl::from(impl).getConfidenceThreshold();
+}
+
+TextDetectionModel_EAST& TextDetectionModel_EAST::setNMSThreshold(float nmsThreshold)
+{
+    TextDetectionModel_EAST_Impl::from(impl).setNMSThreshold(nmsThreshold);
+    return *this;
+}
+float TextDetectionModel_EAST::getNMSThreshold() const
+{
+    return TextDetectionModel_EAST_Impl::from(impl).getNMSThreshold();
+}
 
-    // Threshold
-    Mat bitmap;
-    threshold(binary, bitmap, binThresh, 255, THRESH_BINARY);
 
-    // Scale ratio
-    float scaleHeight = (float)(frame.rows()) / (float)(binary.size[0]);
-    float scaleWidth = (float)(frame.cols()) / (float)(binary.size[1]);
 
-    // Find contours
-    std::vector<std::vector<Point>> contours;
-    bitmap.convertTo(bitmap, CV_8UC1);
-    findContours(bitmap, contours, RETR_LIST, CHAIN_APPROX_SIMPLE);
+struct TextDetectionModel_DB_Impl : public TextDetectionModel_Impl
+{
+    float binaryThreshold;
+    float polygonThreshold;
+    double unclipRatio;
+    int maxCandidates;
+
+    TextDetectionModel_DB_Impl()
+        : binaryThreshold(0.3f)
+        , polygonThreshold(0.5f)
+        , unclipRatio(2.0f)
+        , maxCandidates(0)
+    {
+        CV_TRACE_FUNCTION();
+    }
 
-    // Candidate number limitation
-    size_t numCandidate = 0;
-    if (contours.size() < maxCandidates) {
-        numCandidate = contours.size();
-    } else {
-        numCandidate = maxCandidates;
+    TextDetectionModel_DB_Impl(const Net& network)
+        : TextDetectionModel_DB_Impl()
+    {
+        CV_TRACE_FUNCTION();
+        initNet(network);
     }
 
-    for (size_t i = 0; i < numCandidate; i++) {
-        std::vector<Point> contour = contours[i];
+    void setBinaryThreshold(float binaryThreshold_) { binaryThreshold = binaryThreshold_; }
+    float getBinaryThreshold() const { return binaryThreshold; }
 
-        // Rescale
-        std::vector<Point> contourScaled;
-        for (size_t j = 0; j < contour.size(); j++) {
-            contourScaled.push_back(Point(int(contour[j].x * scaleWidth),
-                                          int(contour[j].y * scaleHeight)));
-        }
+    void setPolygonThreshold(float polygonThreshold_) { polygonThreshold = polygonThreshold_; }
+    float getPolygonThreshold() const { return polygonThreshold; }
+
+    void setUnclipRatio(double unclipRatio_) { unclipRatio = unclipRatio_; }
+    double getUnclipRatio() const { return unclipRatio; }
 
-        // Calculate text contour score
-        if (contourScore(binary, contour) < polyThresh) continue;
+    void setMaxCandidates(int maxCandidates_) { maxCandidates = maxCandidates_; }
+    int getMaxCandidates() const { return maxCandidates; }
 
-        // Unclip
-        std::vector<Point> approx;
-        RotatedRect box = minAreaRect(contourScaled);
-        Point2f vertex[4];
-        box.points(vertex);
-        for (int j = 0; j < 4; j++) {
-            approx.push_back(Point((int)(vertex[3-j].x), (int)(vertex[3-j].y)));
```

Comment:
> 3-j

Please check this part of code too (do we really need to reorder points?)

Reply:
I have removed it, and updated the `rotateVec` in `unclip`.